app/jobs/transfer_basespace_files.rb

```diff
@@ -18,15 +18,17 @@ def self.perform(sample_id, basespace_dataset_id, basespace_access_token)
     if samples_remaining.empty?
       begin
         BasespaceHelper.revoke_access_token(basespace_access_token)
-      rescue
+      rescue HttpHelper::HttpError => e
         # If revoke_access_token is called multiple times, it will fail on the subsequent calls. This is okay.
-        nil
+        if e.status_code == 401
+          Rails.logger.warn("Revoke access token failed. Likely called multiple times by samples sharing the same token.")
+        else
+          raise e
```

Comment:
I noticed you removed a rescue block in the end of this method that was logging the error and ignoring it (which would eventually make it silently fail).

This new version (where you are re-raising the original error) is better, since it propagates the error condition, but it is important to guarantee that the caller method can actually handle these errors. I don't know where exactly this method is being invoked. Have you checked that?